cda-sovd/src/sovd/error.rs

```diff
@@ -64,7 +66,7 @@ impl From<DiagServiceError> for ApiError {
             | DiagServiceError::Nack(_)
             | DiagServiceError::InvalidSession(_)
             | DiagServiceError::UnknownOperation
-            | DiagServiceError::UnexpectedResponse
+            | DiagServiceError::UnexpectedResponse(_)
```

Comment:
suggestion: Imho 502 Bad Gateway would be more appropriate here
> The server was acting as a gateway or proxy and received an invalid response from the upstream server.

On the other hand the same is true for a bunch of other error messages here, so probably ok to keep it like this.